packages/forms/src/Components/BaseFileUpload.php

```diff
@@ -195,6 +195,7 @@ public function callAfterStateUpdated(): static
 
             $this->evaluate($callback, [
                 'state' => $this->isMultiple() ? $state : Arr::first($state ?? []),
```

Comment:
You see here how we decide whether or not the upload is multiple, and if it isnt then call `Arr::first()`? I think it's similar for inside `getOldState()`